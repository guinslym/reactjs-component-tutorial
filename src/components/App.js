import  React from 'react';
import {Testimonial} from '../components/Testimonial';

class App extends React.Component {
  constructor(props){
    super(props);
    this.state = {'data': ""};
  render() {
    return (
      <div className="container">
        <h1 className="text-center">ReactJs SelectField Tutorial</h1>


        <div className="row">

  			</div>

      </div>
    );
  }
}


export default App;
