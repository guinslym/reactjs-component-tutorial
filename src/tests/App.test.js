import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from '../components/App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

const data =  [
{
  name: "Matthew Horizonte",
  jobs: "Web Developer",
  country: 'Australia',
  category: 'testimonial testimonial-danger'
},
{
  name: "Kirkwood John",
  jobs: "Dentist",
  country: 'Germany',
  category: 'testimonial testimonial-warning'
},
{
  name: "Mt. Tallac",
  jobs: "Bus Driver",
  country: 'Canada',
  category: 'testimonial testimonial-primary'
},
{
  name: "John Fitz Oscar",
  jobs: 34,
  country: 'New Zeland',
  category: 'testimonial testimonial-default'
}
];

test("App snapchot test", () => {
	//create a component that hold our rendered app
	const component = renderer.create(<App />);
	//transform our component to JSON
	const tree = component.toJSON();
	// expect both to be the same
	expect(tree).toMatchSnapshot();
});

describe("Number test", () =>{
	// Number
	const numItems = data.length;

	test("Number of items == 4", () => {
		expect(numItems).toBe(4);
	});

	test("Number of items to be > 2", () => {
		expect(numItems).toBeGreaterThanOrEqual(2);
	});

});

describe("Strings Test", () => {
	// Strings
	const dataTest = data[0].jobs;


	test("There is a dev in the Jobs title", () => {
		expect(dataTest).toMatch(/Deve*/);
	});

	test("The title contains React", () => {
		expect(dataTest).toContain("Web Developer");
	});
});

/*
describe("Arrays Test", () =>{
	// Arrays
	const data2 = ["React Native", "MeteorJS"];

	test("the list of of Name, () =>{
		expect(["React Native", "MeteorJS", "React"]).toEqual(expect.arrayContaining(data2));
	});
});
*/


describe("Objects Tests", () => {
	// Objects

	test("the first item has a property named 'name' ", () =>{
		expect(data[0]).toHaveProperty('name');
	});

	test("the first item has a property named 'name' with a value of Horizonte", () =>{
		expect(data[0]).toHaveProperty('name', "Matthew Horizonte");
	});
});
