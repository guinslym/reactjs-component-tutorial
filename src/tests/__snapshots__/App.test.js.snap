// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapchot test 1`] = `
<div
  className="container"
>
  <h1
    className="text-center"
  >
    ReactJs Multiple Components Tutorial
  </h1>
  <div
    className="row"
  >
    <div
      className="col-sm-6"
    >
      <div
        className="testimonial testimonial-danger"
        id="tb-testimonial"
      >
        <div
          className="testimonial-section"
        >
          Denim you probably haven't heard of. Lorem ipsum dolor met consectetur adipisicing sit amet, consectetur adipisicing elit, of them jean shorts sed magna aliqua. Lorem ipsum dolor met.
        </div>
        <div
          className="testimonial-desc"
        >
          <img
            alt=""
            src="https://placeholdit.imgix.net/~text?txtsize=9&txt=100%C3%97100&w=100&h=100"
          />
          <div
            className="testimonial-writer"
          >
            <div
              className="testimonial-writer-name"
            >
              Matthew Horizonte
            </div>
            <div
              className="testimonial-writer-designation"
            >
              Web Developer
            </div>
            <a
              className="testimonial-writer-company"
              href="#"
            >
              Australia
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-sm-6"
    >
      <div
        className="testimonial testimonial-warning"
        id="tb-testimonial"
      >
        <div
          className="testimonial-section"
        >
          Denim you probably haven't heard of. Lorem ipsum dolor met consectetur adipisicing sit amet, consectetur adipisicing elit, of them jean shorts sed magna aliqua. Lorem ipsum dolor met.
        </div>
        <div
          className="testimonial-desc"
        >
          <img
            alt=""
            src="https://placeholdit.imgix.net/~text?txtsize=9&txt=100%C3%97100&w=100&h=100"
          />
          <div
            className="testimonial-writer"
          >
            <div
              className="testimonial-writer-name"
            >
              Kirkwood John
            </div>
            <div
              className="testimonial-writer-designation"
            >
              Dentist
            </div>
            <a
              className="testimonial-writer-company"
              href="#"
            >
              Germany
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-sm-6"
    >
      <div
        className="testimonial testimonial-primary"
        id="tb-testimonial"
      >
        <div
          className="testimonial-section"
        >
          Denim you probably haven't heard of. Lorem ipsum dolor met consectetur adipisicing sit amet, consectetur adipisicing elit, of them jean shorts sed magna aliqua. Lorem ipsum dolor met.
        </div>
        <div
          className="testimonial-desc"
        >
          <img
            alt=""
            src="https://placeholdit.imgix.net/~text?txtsize=9&txt=100%C3%97100&w=100&h=100"
          />
          <div
            className="testimonial-writer"
          >
            <div
              className="testimonial-writer-name"
            >
              Mt. Tallac
            </div>
            <div
              className="testimonial-writer-designation"
            >
              Bus Driver
            </div>
            <a
              className="testimonial-writer-company"
              href="#"
            >
              Canada
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-sm-6"
    >
      <div
        className="testimonial testimonial-default"
        id="tb-testimonial"
      >
        <div
          className="testimonial-section"
        >
          Denim you probably haven't heard of. Lorem ipsum dolor met consectetur adipisicing sit amet, consectetur adipisicing elit, of them jean shorts sed magna aliqua. Lorem ipsum dolor met.
        </div>
        <div
          className="testimonial-desc"
        >
          <img
            alt=""
            src="https://placeholdit.imgix.net/~text?txtsize=9&txt=100%C3%97100&w=100&h=100"
          />
          <div
            className="testimonial-writer"
          >
            <div
              className="testimonial-writer-name"
            >
              John Fitz Oscar
            </div>
            <div
              className="testimonial-writer-designation"
            >
              34
            </div>
            <a
              className="testimonial-writer-company"
              href="#"
            >
              New Zeland
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
